pyfesom2/transport.py

```diff
@@ -0,0 +1,1083 @@
+                           'lon_end': 10.6,
+                           'lat_start': 78.8,
+                           'lat_end': 78.8,
+                           }
+            elif section_name == 'FRAMSTRAIT':
+                section = {'lon_start': -6,
+                           'lon_end': 10.6,
+                           'lat_start': 78.8,
+                           'lat_end': 78.8,
+                           }
+
+            elif section_name == 'ST_ANNA_TROUGH':
+                section = {'lon_start': 60,
+                           'lon_end': 80,
+                           'lat_start': 80,
+                           'lat_end': 80,
+                           }
+
+            elif section_name == 'BS_40E':
+                section = {'lon_start': 40,
+                           'lon_end': 40,
+                           'lat_start': 68,
+                           'lat_end': 80,
+                           }
+
+            # add more presets here
+
+        section['name'] = section_name
+
+    # Find the orientation of the section and look for the nesseccary velocity files
+    if section['lon_start'] == section['lon_end']:
+        section['orientation'] = 'meridional'
+
+    elif (section['lat_start'] == section['lat_end']) :
+        section['orientation'] = 'zonal'
+
+    else:
+        section['orientation'] = 'other'
+        raise ValueError('Only zonal or meridional are currently supported!')
+
+    # Check for existance of the files
+    files_u = [data_path + 'u.fesom.' + str(year) + '.nc' for year in years]
+    files_v = [data_path + 'v.fesom.' + str(year) + '.nc' for year in years]
+
+    files = files_u + files_v
+
+    file_check = []
+    for file in files:
+        file_check.append(isfile(file))
+
+    if not all(file_check):
+        raise FileExistsError('One or more of the velocity files do not exist!')
+
+
+    return mesh, mesh_diag, section, files
+
+def _ComputeWaypoints(section, mesh, use_great_circle):
+    '''
+    compute_waypoints.py
+
+    Computes the waypoints between the section start and end either along a great circle or linear
+
+    Inputs
+    ------
+    section (dict)
+        section dictionary containing the section start, end and orientation data
+    mesh (fesom.mesh object)
+        fesom.mesh
+    use_great_circle (bool)
+        True or False
+
+
+    Returns
+    -------
+    section waypoints ()
+        waypoints along the section
+    mesh (fesom.mesh object)
+        fesom mesh
+    section (dict)
+        dictionary containing section information
+    '''
+
+    if use_great_circle:
+        # Compute the great circle coordinates along the section
+        g = pyproj.Geod(ellps='WGS84')
+
+        section_waypoints = g.npts(section['lon_start'],
+                                   section['lat_start'],
+                                   section['lon_end'],
+                                   section['lat_end'],
+                                   1000
```

Comment:
Why not to make number of points optional? For larger meshes extracting 1000 points took be  very long, and you don't need so much on coarser resolution meshes.